pkg/cosign/tlog.go

```diff
@@ -184,17 +184,13 @@ func doUpload(ctx context.Context, rekorClient *client.Rekor, pe models.Proposed
 	return nil, errors.New("bad response from server")
 }
 
-func rekorEntry(payload, signature, pubKey []byte) hashedrekord_v001.V001Entry {
-	// TODO: Signatures created on a digest using a hash algorithm other than SHA256 will fail
-	// upload right now. Plumb information on the hash algorithm used when signing from the
-	// SignerVerifier to use for the HashedRekordObj.Data.Hash.Algorithm.
-	h := sha256.Sum256(payload)
+func rekorEntry(sha256CheckSum, signature, pubKey []byte) hashedrekord_v001.V001Entry {
```

Comment:
I wonder if you want to change `sha256CheckSum []byte` to `checksum hash.Hash` just to make it harder to pass in the full payload (and not the hash output) by accident.

Reply:
I'm a littler nervous that you didn't fix *every* `TLogUpload` call, and doing that would help!

Reply:
Done.